src/control/cmd/dmg/pool.go

```diff
@@ -223,6 +223,114 @@ func (cmd *PoolCreateCmd) Execute(args []string) error {
 		return errors.New("either --size or --scm-size must be set")
 	}
 
+	return nil
+}
+
+func ratio2Percentage(log logging.Logger, scm, nvme float64) (p float64) {
+	p = 100.00
+	min := storage.MinScmToNVMeRatio * p
+
+	if nvme > 0 {
+		p *= scm / nvme
+		if p < min {
+			log.Noticef("SCM:NVMe ratio is less than %0.2f%%, DAOS performance "+
+				"will suffer!", min)
+		}
+		return
+	}
+
+	log.Notice("Creating DAOS pool without NVME storage")
+	return
+}
+
+func (cmd *PoolCreateCmd) storageAutoPercentage(ctx context.Context, req *control.PoolCreateReq) error {
+	if cmd.NumRanks > 0 {
+		return errIncompatFlags("size", "nranks")
+	}
+	if cmd.TierRatio.IsSet() {
+		return errIncompatFlags("size=%", "tier-ratio")
+	}
+
+	scmBytes, nvmeBytes, err := control.GetMaxPoolSize(ctx, cmd, cmd.ctlInvoker,
+		ranklist.RankList(req.Ranks))
+	if err != nil {
+		return err
+	}
+
+	availRatio := cmd.Size.availRatio
+	if availRatio < 100 {
+		scmBytes = availRatio * scmBytes / 100
+		nvmeBytes = availRatio * nvmeBytes / 100
+	}
+	req.TierBytes = []uint64{scmBytes, nvmeBytes}
```

Comment:
Ideally all of this logic should be moved behind the `control.PoolCreate()` API so that non-dmg users can take advantage of it.

Reply:
I tried doing that and ran into an issue which I now cannot remember, I will try again and report back as to why it was a problem.

Reply:
From what I remember, the pool max size calculation was integrated this way for not being too much intrusive in the code.
However, I perfectly OK to move it behind `control.PoolCreate()`: it will probably needs some slight modifications of the `PoolCreateReq` structure.